addons/sourcemod/scripting/KnockbackRestrict.sp

```diff
@@ -445,83 +437,75 @@ void OnGetKbansNumber(Database db, DBResultSet results, const char[] error, int
 			continue;
 		}
 
-		if(CheckCommandAccess(i, "sm_admin", ADMFLAG_BAN, true)) {
+		if(CheckCommandAccess(i, "sm_admin", ADMFLAG_KICK, true)) {
 			CPrintToChat(i, "%t", "PlayerConnect", client, count);
 		}
 	}
 
 	g_iClientKbansNumber[client] = count;
 }
 
-public void OnClientConnected(int client) {
-	if(IsFakeClient(client) || IsClientSourceTV(client)) {
-		return;
-	}
-
-	if(g_bIsClientRestricted[client]) {
-		return;
-	}
+public Action Event_OnPlayerName(Handle event, const char[] name, bool dontBroadcast)
+{
+	int client = GetClientOfUserId(GetEventInt(event, "userid"));
+	if (client > 0 && IsClientInGame(client) && !IsFakeClient(client) && !IsClientSourceTV(client))
+		GetEventString(event, "newname", g_sName[client], sizeof(g_sName[]));
```

Comment:
sizeof(g_sName[client])

Reply:
No it will throw this error.
`error 001: expected token: "]", but found "-identifier-"`

Reply:
okay